src/test/util/matchable.rs

```diff
@@ -59,6 +59,20 @@ impl Matchable for Bson {
 impl Matchable for Document {
     fn content_matches(&self, expected: &Document) -> bool {
         for (k, v) in expected.iter() {
+            if k == "upsertedCount" {
```

Comment:
According to the [crud spec](https://github.com/mongodb/specifications/tree/master/source/crud/tests#optional-fields-in-expected-result-objects), if we don't include an upsertedCount filed in our result, we can just skip verifying it in the test runner.